src/package-hoister.js

```diff
@@ -142,15 +146,15 @@ export default class PackageHoister {
     //
     let parentParts: Parts = [];
     const isIncompatible = ref.incompatible;
-    let isRequired = isDirectRequire && !ref.ignore && !isIncompatible;
+    let isRequired = isDirectRequire && !ref.ignore && !isIncompatible && (!ref.optional || !this.ignoreOptional);
```

Comment:
Better to write as `!(ref.optional && this.ignoreOptional)` since makes the intent clearer. Also possibly fewer instructions :P